### PR TITLE
refactor: remove `QueryChunkMeta::timestamp_min_max`

### PR DESCRIPTION
**Pull request by crepererum**
On top of #5961.

Use the table summary instead. This allows us to have a single mechanism
that both IOx and DataFusion understand. This basically lifts the "basic
table summary" mechanism that the querier uses to `iox_query` and let
the compactor and ingester use the same mechanism.

While not strictly necessary, simplifying the `QueryChunk[Meta]`
interface helps with #5897.

### CODE REVIEW

**Review by NGA-TRAN — APPROVED** [review]
I like this refactor. Using TableSummary everywhere is definitely better

**Comment by NGA-TRAN** [comment]
But btw, this will conflict with @Dom's recent ingested refactoring that may not be merged soon.

**Comment by alamb** [comment]
This is a good simplification -- thank you 👍

File: ingester/src/query_adaptor.rs [hunk 1]
```diff
@@ -162,24 +180,14 @@ impl QueryChunkMeta for QueryAdaptor {
         None // Ingester data is not sorted
     }
 
-    fn timestamp_min_max(&self) -> Option<TimestampMinMax> {
```
Comment (alamb) [inline ingester/src/query_adaptor.rs:165]: Thank you -- this is a good cleanup

**Pull request merged**